### PR TITLE
Add memory arbitration time runtime stats for operator

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Add runtime stats to track the memory arbitration time triggered by an operator.
This helps performance debug how much time a long running operator spent
on memory arbitration.
The followup is to collect memory reclaim times in task stats and report as a task
level runtime stats by Prestissimo.

### CODE REVIEW

File: velox/exec/SharedArbitrator.cpp [hunk 1]
```diff
@@ -575,12 +576,16 @@ SharedArbitrator::ScopedArbitration::ScopedArbitration(
...
       std::chrono::duration_cast<std::chrono::microseconds>(
-          std::chrono::steady_clock::now() - startTime_);
+          std::chrono::steady_clock::now() - startTime_)
+          .count();
   RECORD_HISTOGRAM_METRIC_VALUE(
-      kMetricArbitratorArbitrationTimeMs, arbitrationTime.count() / 1'000);
-  arbitrator_->arbitrationTimeUs_ += arbitrationTime.count();
+      kMetricArbitratorArbitrationTimeMs, arbitrationTimeUs / 1'000);
+  addThreadLocalRuntimeStat(
+      "memoryArbitrationTime",
```
Comment (mbasmanova) [inline velox/exec/SharedArbitrator.cpp:586]: Other similar metrics use xxxNanos or xxxWallNanos naming. Perhaps, rename to memoryArbitrationNanos

**Review by mbasmanova — COMMENTED** [review]
@xiaoxmeng Thank you for adding this stat. Would you update the PR description to explain what it is tracking? I assume it reports wall time in nanoseconds of how long memory arbitration took and it is attached to the operator that triggered memory arbitration.

Is it possible for arbitration to be triggered by query 1, but arbitration itself to pause and reclaim memory from query 2. In this case, will we only see arbitration time in stats of query 1? If so, can we make it so that arbitration stats are available from all queries that are affected?

**Comment by xiaoxmeng** [comment]
> Is it possible for arbitration to be triggered by query 1, but arbitration itself to pause and reclaim memory from query 2. In this case, will we only see arbitration time in stats of query 1? If so, can we make it so that arbitration stats are available from all queries that are affected?

Yeah, it is possible and those queries are stopped and not counted in their cpu times. We could add runtime stats to track task reclaim time as we discussed yesterday but currently task level runtime stats is per pipeline.

File: velox/exec/SharedArbitrator.cpp (hunk 1 above)
Comment (xiaoxmeng) [reply velox/exec/SharedArbitrator.cpp:586]: Updated. Thanks!

**Review by mbasmanova — APPROVED** [review]
Thanks.

**Pull request closed without merging**